File: CS/AddComboItemPopup/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using DevExpress.Web.Mvc;

namespace AddComboItemPopup.Controllers {
    public class HomeController : Controller {
        public static OrderItemsProvider orderItemsProvider;

        public ActionResult Index() {
            orderItemsProvider = new OrderItemsProvider();
            return GridView();
        }
        public ActionResult GridView() {
            ViewData["Products"] = orderItemsProvider.GetProducts();
            return View("Index", orderItemsProvider.GetItems());
        }
        public ActionResult GridViewPartial() {
            ViewData["Products"] = orderItemsProvider.GetProducts();
            return PartialView("_GridView", orderItemsProvider.GetItems());
        }

        public ActionResult gridComboBoxPartial() {
            return PartialView("_EditComboBox", orderItemsProvider.GetProducts());
        }

        #region Main grid ppdate procedures
        [HttpPost, ValidateInput(false)]
        public ActionResult AddNewPartial([ModelBinder(typeof(DevExpressEditorsBinder))] OrderItem item) {
            if (ModelState.IsValid) {
                try {
                    orderItemsProvider.ItemInsert(item.ProductId, item.Quantity);
                }
                catch (Exception e) {
                    ViewData["EditError"] = e.Message;
                }
            }
            else
                ViewData["EditError"] = ModelState["ProductId"].Errors[0].ErrorMessage;

            ViewData["Products"] = orderItemsProvider.GetProducts();
            return PartialView("_GridView", orderItemsProvider.GetItems());
        }
        [HttpPost, ValidateInput(false)]
        public ActionResult UpdatePartial([ModelBinder(typeof(DevExpressEditorsBinder))] OrderItem item) {
            if (ModelState.IsValid) {
                try {
                    orderItemsProvider.ItemUpdate(item.ProductId, item.Quantity, item.Id);
                }
                catch (Exception e) {
                    ViewData["EditError"] = e.Message;
                }
            }
            else
                ViewData["EditError"] = ModelState["ProductId"].Errors[0].ErrorMessage;

            ViewData["Products"] = orderItemsProvider.GetProducts();
            return PartialView("_GridView", orderItemsProvider.GetItems());
        }
        [HttpPost, ValidateInput(false)]
        public ActionResult DeletePartial(int Id) {
            if (Id >= 0) {
                try {
                    orderItemsProvider.ItemDelete(Id);
                }
                catch (Exception e) {
                    ViewData["EditError"] = e.Message;
                }
            }
            return PartialView("_GridView", orderItemsProvider.GetItems());
        }
        #endregion

        #region ComboBox grid ppdate procedures
        [HttpPost, ValidateInput(false)]
        public ActionResult AddNewComboBoxPartial([ModelBinder(typeof(DevExpressEditorsBinder))] OrderProduct product) {
            if (ModelState.IsValid) {
                try {
                    orderItemsProvider.ProductInsert(product.Name, product.Price);
                }
                catch (Exception e) {
                    ViewData["EditgridComboBoxError"] = e.Message;
                }
            }
            else
                ViewData["EditgridComboBoxError"] = ModelState["Id"].Errors[0].ErrorMessage;

            return PartialView("_EditComboBox", orderItemsProvider.GetProducts());
        }
        [HttpPost, ValidateInput(false)]
        public ActionResult UpdateComboBoxPartial([ModelBinder(typeof(DevExpressEditorsBinder))] OrderProduct product) {
            if (ModelState.IsValid) {
                try {
                    orderItemsProvider.ProductUpdate(product.Name, product.Price, product.Id);
                }
                catch (Exception e) {
                    ViewData["EditgridComboBoxError"] = e.Message;
                }
            }
            else
                ViewData["EditgridComboBoxError"] = ModelState["Id"].Errors[0].ErrorMessage;

            return PartialView("_EditComboBox", orderItemsProvider.GetProducts());
        }

        #endregion
    }
}

File: VB/AddComboItemPopup/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc

Namespace AddComboItemPopup.Controllers
	Public Class HomeController
		Inherits Controller
		Public Shared orderItemsProvider As OrderItemsProvider

		Public Function Index() As ActionResult
			orderItemsProvider = New OrderItemsProvider()
			Return GridView()
		End Function
		Public Function GridView() As ActionResult
			ViewData("Products") = orderItemsProvider.GetProducts()
			Return View("Index", orderItemsProvider.GetItems())
		End Function
		Public Function GridViewPartial() As ActionResult
			ViewData("Products") = orderItemsProvider.GetProducts()
			Return PartialView("_GridView", orderItemsProvider.GetItems())
		End Function

		Public Function gridComboBoxPartial() As ActionResult
			Return PartialView("_EditComboBox", orderItemsProvider.GetProducts())
		End Function

		#Region "Main grid ppdate procedures"
		<HttpPost, ValidateInput(False)> _
		Public Function AddNewPartial(<ModelBinder(GetType(DevExpressEditorsBinder))> ByVal item As OrderItem) As ActionResult
			If ModelState.IsValid Then
				Try
					orderItemsProvider.ItemInsert(item.ProductId, item.Quantity)
				Catch e As Exception
					ViewData("EditError") = e.Message
				End Try
			Else
				ViewData("EditError") = ModelState("ProductId").Errors(0).ErrorMessage
			End If

			ViewData("Products") = orderItemsProvider.GetProducts()
			Return PartialView("_GridView", orderItemsProvider.GetItems())
		End Function
		<HttpPost, ValidateInput(False)> _
		Public Function UpdatePartial(<ModelBinder(GetType(DevExpressEditorsBinder))> ByVal item As OrderItem) As ActionResult
			If ModelState.IsValid Then
				Try
					orderItemsProvider.ItemUpdate(item.ProductId, item.Quantity, item.Id)
				Catch e As Exception
					ViewData("EditError") = e.Message
				End Try
			Else
				ViewData("EditError") = ModelState("ProductId").Errors(0).ErrorMessage
			End If

			ViewData("Products") = orderItemsProvider.GetProducts()
			Return PartialView("_GridView", orderItemsProvider.GetItems())
		End Function
		<HttpPost, ValidateInput(False)> _
		Public Function DeletePartial(ByVal Id As Integer) As ActionResult
			If Id >= 0 Then
				Try
					orderItemsProvider.ItemDelete(Id)
				Catch e As Exception
					ViewData("EditError") = e.Message
				End Try
			End If
			Return PartialView("_GridView", orderItemsProvider.GetItems())
		End Function
		#End Region

		#Region "ComboBox grid ppdate procedures"
		<HttpPost, ValidateInput(False)> _
		Public Function AddNewComboBoxPartial(<ModelBinder(GetType(DevExpressEditorsBinder))> ByVal product As OrderProduct) As ActionResult
			If ModelState.IsValid Then
				Try
					orderItemsProvider.ProductInsert(product.Name, product.Price)
				Catch e As Exception
					ViewData("EditgridComboBoxError") = e.Message
				End Try
			Else
				ViewData("EditgridComboBoxError") = ModelState("Id").Errors(0).ErrorMessage
			End If

			Return PartialView("_EditComboBox", orderItemsProvider.GetProducts())
		End Function
		<HttpPost, ValidateInput(False)> _
		Public Function UpdateComboBoxPartial(<ModelBinder(GetType(DevExpressEditorsBinder))> ByVal product As OrderProduct) As ActionResult
			If ModelState.IsValid Then
				Try
					orderItemsProvider.ProductUpdate(product.Name, product.Price, product.Id)
				Catch e As Exception
					ViewData("EditgridComboBoxError") = e.Message
				End Try
			Else
				ViewData("EditgridComboBoxError") = ModelState("Id").Errors(0).ErrorMessage
			End If

			Return PartialView("_EditComboBox", orderItemsProvider.GetProducts())
		End Function

		#End Region
	End Class
End Namespace
